src/test/shared/utilities/vsCodeUtils.test.ts

```diff
@@ -165,7 +165,7 @@ describe('vsCodeUtils getChannelLogger', function() {
         testDataCase
     }: TestCaseParams) => {
         // Log message to channel
-        channelLogger[logLevel](
+        (channelLogger as unknown as { [logLevel: string]: TemplateHandler })[logLevel](
```

Comment:
Just add the index signature to `ChannelLogger`, i.e.:

```typescript
export interface ChannelLogger {
    verbose: TemplateHandler
    debug: TemplateHandler
    info: TemplateHandler
    warn: TemplateHandler
    error: TemplateHandler

    [logLevel: keyof ChannelLogger]: TemplateHandler
}
```

Reply:
An indexing signature cannot be a union type. Code works as is.